### PR TITLE
Ciac 12050 ews extension online powershell v3 add commands

### PR DESCRIPTION
**Pull request by ilaredo**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-12050](https://jira-dc.paloaltonetworks.com/browse/CIAC-12050)

## Description
add commands:
ews-rule-disable
ews-rule-enable
ews-mail-flow-rules-list
ews-mail-flow-rule-get
ews-mail-flow-rule-remove
ews-mail-flow-rule-disable
ews-mail-flow-rule-enable
ews-mail-forwarding-disable

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.yml [hunk 1]
```diff
@@ -2402,6 +2402,103 @@ script:
       description: The Inbox rule that you want to remove.
       required: true
     description: Remove an Inbox rule.
+  - name: ews-rule-disable
+    arguments:
+      - name: mailbox
+        description: The mailbox that contains the Inbox rule.
+        required: true
+      - name: identity
+        description: The Inbox rule that you want to disable.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.yml:2411]:
```suggestion
        description: The mailbox that contains the inbox rule.
        required: true
      - name: identity
        description: The inbox rule that you want to disable.
```

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.yml [hunk 2]
```diff
@@ -2402,6 +2402,103 @@ script:
...
+        required: true
+    description: Disable an existing inbox rule in a given mailbox.
+  - name: ews-rule-enable
+    arguments:
+      - name: mailbox
+        description: The mailbox that contains the Inbox rule.
+        required: true
+      - name: identity
+        description: The Inbox rule that you want to enable.
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.yml:2420]:
```suggestion
        description: The mailbox that contains the inbox rule.
        required: true
      - name: identity
        description: The inbox rule that you want to enable.
```

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md [hunk 3]
```diff
@@ -1793,3 +1793,214 @@ There are no context outputs for this command.
...
+
+#### Base Command
+
+`ews-rule-disable`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| mailbox | The mailbox that contains the Inbox rule. | Required | 
+| identity | The Inbox rule that you want to disable. | Required |
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md:1809]:
```suggestion
| mailbox | The mailbox that contains the inbox rule. | Required | 
| identity | The inbox rule that you want to disable. | Required | 
```

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md [hunk 4]
```diff
@@ -1793,3 +1793,214 @@ There are no context outputs for this command.
...
+
+#### Base Command
+
+`ews-rule-enable`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| mailbox | The mailbox that contains the Inbox rule. | Required | 
+| identity | The Inbox rule that you want to enable. | Required |
```
Comment (ShirleyDenkberg) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md:1831]:
```suggestion
| mailbox | The mailbox that contains the inbox rule. | Required | 
| identity | The inbox rule that you want to enable. | Required | 
```

**Comment by ShirleyDenkberg** [comment]
@amshamah419 @thefrieddan1 @eyalpalo Doc review completed.

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1 [hunk 5]
```diff
@@ -1417,6 +1491,290 @@ class ExchangeOnlinePowershellV3Client
...
+    }
+
+    [PSObject]ListMailFlowRules()
+    {
+        $response = ""
+        try {
+            # Establish session to remote
+            $this.CreateSession()
+            # Import and Execute command
+            $response = Get-TransportRule -WarningAction:SilentlyContinue
```
Comment (eyalpalo) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1:1589]: We might want to add a limit in list commands - so we won't bring all results if there are many. Is the API support limit/pagination?

**Review by eyalpalo — CHANGES_REQUESTED** [review]
Nice! Add RN and please assign @ShirleyDenkberg for doc review

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1 (hunk 5 above)
Comment (ilaredo) [reply Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/EwsExtensionEXOPowershellV3.ps1]: added

File: Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md [hunk 6]
```diff
@@ -1846,9 +1846,10 @@ List all mail flow rules (transport rules) in the organization.
 
 `ews-mail-flow-rules-list`
 #### Input
-| **Argument Name** | **Description**                                                | **Possible Values** | **Is Array** | **Required** | **Note** |
-| --- |----------------------------------------------------------------|---------------------| --- | --- | --- |
-| extended_output | Determine whether the output will be in verbose format or not. | Boolean             | No | No | Default = False |
+| **Argument Name** | **Description**                                                | **Possible Values** | **Is Array** | **Required** | **Note**        |
+|-------------------|----------------------------------------------------------------|---------------------|--------------| --- |-----------------|
+| extended_output   | Determine whether the output will be in verbose format or not. | Boolean             | No           | No | Default = False |
+| limit             | The amount of mail flow rules to return. | Number             | No           | No | Default = 1000  |
```
Comment (eyalpalo) [inline Packs/MicrosoftExchangeOnline/Integrations/EwsExtensionEXOPowershellV3/README.md:1852]:
```suggestion
| limit             | The amount of mail flow rules to return. | Number             | No           | No | Default is 1000  |
```

**Review by eyalpalo — APPROVED** [review]
Just validate that the new limit arg is working as expected when passed and when not.

**Pull request merged**